### PR TITLE
(CONT-797) - Support RedHat Satellite 6.13

### PR DESCRIPTION
**Pull request by Ramesh7**
- Enabling Puppet 8
- Updating Satellite to 6.13 
- Updating deprecated function & references
- CI pipeline fixes

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,56 +5,67 @@ on:
...
+
+    - name: Activate Ruby 2.7
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: "2.7"
+        bundler-cache: true
+
+    - name: Bundle Install
       run: |
         gem install bundler
```
Comment (jordanbreen28) [inline .github/workflows/ci.yml:33]: I don't think the `gem install bundler` is needed, bundler is installed in the `Activate Ruby 2.7` step above

File: request.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"url":"https://api.github.com/repos/puppetlabs/puppetlabs-satellite_pe_tools/actions/runs/5251671377","VMs":[{"cloud":null,"region":null,"zone":null,"images":["centos-stream-8"]}]}
```
Comment (jordanbreen28) [inline request.json:1]: this file made it in by mistake

File: spec/acceptance/test_spec.rb [hunk 3]
```diff
@@ -38,8 +37,8 @@ class {'satellite_pe_tools':
       Helper.instance.run_shell('puppet agent -t', expect_failures: true)
     end
 
-    it 'contains the report text in Satellite' do
-      expect(satellite_get_last_report(satellite_host, server_host).to_s).to match(%r{This is a test from Puppet to Satellite})
+    xit 'contains the report text in Satellite' do
```
Comment (jordanbreen28) [inline spec/acceptance/test_spec.rb:40]: typo
```suggestion
    it 'contains the report text in Satellite' do
```

Comment (Ramesh7) [reply spec/acceptance/test_spec.rb:40]: its intentional change, I have commented this change as I mentioned on slack.
Not sure if you have idea but this is how you can move the spec into pending.

File: request.json (hunk 2 above)
Comment (Ramesh7) [reply request.json:1]: yeah, its by mistake. Thanks for eagle eyes.

File: spec/acceptance/test_spec.rb (hunk 3 above)
Comment (jordanbreen28) [reply spec/acceptance/test_spec.rb:40]: AH! we would also opt for this way, so we can add a reason why also.
https://github.com/puppetlabs/puppetlabs-node_encrypt/blob/ab6e46cd6fc9c14a3297df89b2ec690e026828ed/spec/defines/file_spec.rb#L38

**Review by jordanbreen28 — APPROVED** [review]

**Pull request merged**